lib/ansible/modules/cloud/amazon/aws_ssm_send_command.py

```diff
@@ -0,0 +1,212 @@
+status:
+    description: Status of the run command.
+    returned: success
+    type: str
+    sample: Success
+'''
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import boto3_conn, get_aws_connection_info, AWSRetry
+import traceback
+from time import sleep
+
+try:
+    import botocore
+except ImportError:
+    pass  # will be captured by imported HAS_BOTO3
+
+
+@AWSRetry.backoff(tries=5, delay=5, backoff=2.0)
+def ssm_send_command(conn, **kwargs):
+    return conn.send_command(**kwargs)
+
+
+@AWSRetry.backoff(tries=5, delay=5, backoff=2.0)
+def ssm_list_command_invocations(conn, **kwargs):
+    return conn.list_command_invocations(**kwargs)
+
+
+def main():
+    argument_spec = dict(
+        name=dict(required=True),
+        wait=dict(default=True, type='bool'),
+        comment=dict(),
+        instance_ids=dict(required=True, type='list'),
+        parameters=dict(default={}, type='dict')
+    )
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        supports_check_mode=False
+    )
+
+    # Needs to be an existing SSM document name
+    document_name = module.params.get('name')
+    comment = module.params.get('comment')
+    await_return = module.params.get('wait')
+    instance_ids = module.params.get('instance_ids')
+    parameters = module.params.get('parameters')
+
+    if not (document_name and instance_ids):
+        module.fail_json(
+            msg="Must provide SSM document name and at least one instance id.")
+
+    conn = module.client('ssm')
+
+    invoke_params = {}
+
+    if document_name:
+        invoke_params['DocumentName'] = document_name
+    if comment:
+        invoke_params['Comment'] = comment
+    if instance_ids:
+        invoke_params['InstanceIds'] = instance_ids
+    if parameters:
+        invoke_params['Parameters'] = parameters
+
+    try:
+        response = ssm_send_command(conn, **invoke_params)
+    except botocore.exceptions.ClientError as ce:
+        if ce.response['Error']['Code'] == 'ResourceNotFoundException':
+            module.fail_json_aws(ce, msg="Could not find the SSM doc to execute. Make sure "
+                                 "the document name is correct and your profile has "
+                                 "permissions to execute SSM.")
+        module.fail_json_aws(
+            ce, msg="Client-side error when invoking SSM, check inputs and specific error")
+    except botocore.exceptions.ParamValidationError as ve:
+        module.fail_json_aws(
+            ve, msg="Parameters to `invoke` failed to validate")
+    except Exception as e:
+        module.fail_json(msg="Unexpected failure while invoking SSM send command.",
+                         exception=traceback.format_exc(e))
+
+    if await_return:
+        command_id = response['Command']['CommandId']
```

Comment:
Perhaps change this to:
`command_id = response.get('Comand', {}).get('CommandId)`
to nicely handle the case in which the received response is malformed for what ever reason. The `if` statement on line 177 will work nicely in this case.